ui/helpers/utils/settings-search.js

```diff
@@ -95,205 +95,218 @@ export function getSettingsRoutes(t) {
       section: t('usePhishingDetection'),
       description: t('usePhishingDetectionDescription'),
       route: `${SECURITY_ROUTE}#phishing-detection`,
       image: 'security-icon.svg',
-      id: 22,
+      id: 23,
     },
     {
       tab: t('securityAndPrivacy'),
       section: t('participateInMetaMetrics'),
       description: t('participateInMetaMetricsDescription'),
       route: `${SECURITY_ROUTE}#metrametrics`,
       image: 'security-icon.svg',
-      id: 23,
+      id: 24,
     },
     {
       tab: t('alerts'),
       section: t('alertSettingsUnconnectedAccount'),
       description: t('alertSettingsUnconnectedAccount'),
       route: `${ALERTS_ROUTE}#unconnected-account`,
       image: 'alerts-icon.svg',
-      id: 24,
+      id: 25,
     },
     {
       tab: t('alerts'),
       section: t('alertSettingsWeb3ShimUsage'),
       description: t('alertSettingsWeb3ShimUsage'),
       route: `${ALERTS_ROUTE}#web3-shimusage`,
       image: 'alerts-icon.svg',
-      id: 25,
+      id: 26,
     },
     {
       tab: t('networks'),
       section: t('mainnet'),
       description: t('mainnet'),
       route: `${NETWORKS_ROUTE}#networks-mainnet`,
       image: 'network-icon.svg',
-      id: 26,
+      id: 27,
     },
     {
       tab: t('networks'),
       section: t('ropsten'),
       description: t('ropsten'),
       route: `${NETWORKS_ROUTE}#networks-ropsten`,
       image: 'network-icon.svg',
-      id: 27,
+      id: 28,
     },
     {
       tab: t('networks'),
       section: t('rinkeby'),
       description: t('rinkeby'),
       route: `${NETWORKS_ROUTE}#networks-rinkeby`,
       image: 'network-icon.svg',
-      id: 28,
+      id: 29,
     },
     {
       tab: t('networks'),
       section: t('goerli'),
       description: t('goerli'),
       route: `${NETWORKS_ROUTE}#networks-goerli`,
       image: 'network-icon.svg',
-      id: 29,
+      id: 30,
     },
     {
       tab: t('networks'),
       section: t('kovan'),
       description: t('kovan'),
       route: `${NETWORKS_ROUTE}#networtks-kovan`,
       image: 'network-icon.svg',
-      id: 30,
+      id: 31,
     },
     {
       tab: t('networks'),
       section: t('localhost'),
       description: t('localhost'),
       route: `${NETWORKS_ROUTE}#network-localhost`,
       image: 'network-icon.svg',
-      id: 31,
-    },
-    {
-      tab: t('experimental'),
-      section: t('useTokenDetection'),
-      description: t('useTokenDetectionDescription'),
-      route: `${EXPERIMENTAL_ROUTE}#token-description`,
-      image: 'experimental-icon.svg',
       id: 32,
     },
+    process.env.TOKEN_DETECTION_V2
+      ? {}
+      : {
+          /** TODO: Remove during TOKEN_DETECTION_V2 feature flag clean up */
+          tab: t('experimental'),
+          section: t('useTokenDetection'),
+          description: t('useTokenDetectionDescription'),
+          route: `${EXPERIMENTAL_ROUTE}#token-description`,
+          image: 'experimental-icon.svg',
+          id: 33,
+        },
```

Comment:
inserting empty object here does not looks great, mey be use array destructuring here to not insert anything if  `process.env.TOKEN_DETECTION_V2` is not enabled.